controllers/twophase/state_machine_test.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2020 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package twophase
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/chaos-mesh/chaos-mesh/pkg/mock"
+	ctx "github.com/chaos-mesh/chaos-mesh/pkg/router/context"
+)
+
+func TestStateMachine(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"Twophase StateMachine Suite",
+		[]Reporter{envtest.NewlineReporter{}})
+}
+
+var _ = Describe("TwoPhase StateMachine", func() {
+	Context("TwoPhase", func() {
+
+		status := []v1alpha1.ExperimentPhase{
+			v1alpha1.ExperimentPhaseFailed,
+			v1alpha1.ExperimentPhaseFinished,
+			v1alpha1.ExperimentPhasePaused,
+			v1alpha1.ExperimentPhaseRunning,
+			v1alpha1.ExperimentPhaseWaiting,
+		}
+
+		It("StateMachine Target Finish", func() {
+			defer mock.With("MockApplyError", errors.New("ApplyError"))()
+			defer mock.With("MockRecoverError", errors.New("RecoverError"))()
+
+			for _, status := range status {
```

Comment:
for status range status ? 
```suggestion
			for _, status := range statuses {
```

Reply:
Fixed